components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -43,7 +45,24 @@ class WebExtensionBrowserMenuBuilder(
         val extensionMenuItems =
             WebExtensionBrowserMenu.getOrUpdateWebExtensionMenuItems(store.state, store.state.selectedTab)
 
-        val webExtMenuItem = if (extensionMenuItems.isNotEmpty()) {
+        val filteredExtensionMenuItems = extensionMenuItems.toMutableList()
+
+        val finalList = items.toMutableList()
+
+        extensionMenuItems.forEach { webExtensionBrowserMenuItem ->
+            val index = finalList.indexOfFirst { browserMenuItem ->
+                (browserMenuItem as? PlaceholderBrowserMenuItem)?.label ==
```

Comment:
it looks like the `PlaceholderBrowserMenuItem.label` contains the id, should we rename it to `id`?